web/js/codeworld_shared.js

```diff
@@ -245,20 +245,14 @@ function renderDeclaration(decl, keyword, keywordData, maxLen, argIndex = -1) {
         }
     }
     if (argIndex >= 0) {
-        if (!leftover.length) {
-            leftover = " :: Could not decipher arguments";
-        }
-        else {
-            let tokens = leftover.substring(
-                leftover.lastIndexOf("(") + 1,
-                leftover.lastIndexOf(")")
-            ).split(",").map(token => token.trim());
-            const ReturnType  = leftover.split("->")[1].trim();
-            argIndex = Math.min(argIndex, tokens.length - 1);
-            tokens[argIndex] = `<strong>${tokens[argIndex]}</strong>`;
-            leftover = ` :: (${tokens.join(', ')}) -> ${ReturnType}`;
-
-        }
+        let tokens = leftover.substring(
```

Comment:
I'm not sure what this is going to do if the declaration looks different from the way you expect.  Another (safer) approach might be to match a regular expression to pull out the arguments, like:

    const [start, args, remainder] = /^(\s*::\s*[(])(.*)([)]\s*->.*)$/.exec(leftover)

You'll also want another pattern for functions of one argument, since there are no parentheses in their types.  (I suspect these don't work in the current code?)

Reply:
Understood. I will work this out in a few hours.

Reply:
Great.  I'm going to bed now, but I'll merge this tomorrow when I have time to do some manual testing.

Reply:
I came up with a regex that works for both single arg and multiple arg declarations 
```javascript
const [head, args, tail] = (/^(\s*::\s*[(]*)([\w,\s]*)([)]*\s*->.*)$/).exec(leftover).slice(1)
```